python/lsst/ap/verify/pipeline_driver.py

```diff
@@ -202,6 +202,23 @@ def runApPipe(metricsJob, workspace, parsedCmdLine):
 
     dataId = _parseDataId(parsedCmdLine.dataId)
     processes = parsedCmdLine.processes
+    
+    # borrowed from validate_drp
+    def extract_instrument_from_butler(butler):
```

Comment:
Please make this a package-private function outside `runApPipe`. The implementation of `runApPipe` will be changing quite a bit in the near future, and the changes will be easier to merge if I don't have to code around a nested function definition.

Reply:
Will do.  Good point.